backend/app/routes/admin/updateAdmin.js

```diff
@@ -4,33 +4,32 @@ const { ErrorHandler } = require('../../../helpers/error');
 const constants = require('../../../constants');
 
 module.exports = async (req, res, next) => {
-  const userRecord = await Admin.findOne({
-    email: res.locals.decode.email,
-  });
-
+  const { firstName, lastName, contact, username } = req.body;
   // Finding user through email which needs to be updated
-  const user = await Admin.findOne({ email: userRecord.email });
 
   // Checking which fields are present and making changes to the present field
-  if (user) {
-    user.username = req.body.username || user.username;
-    user.contact = req.body.contact || user.contact;
-    user.firstName = req.body.firstName || user.firstName;
-    user.lastName = req.body.lastName || user.Name;
-  }
 
   // Updating the document with updated data
-  const [err] = await to(Admin.findOneAndUpdate({ email: userRecord.email }, user));
+  // validation
+  const userbody = {
+    ...(firstName && { firstName }),
+    ...(lastName && { lastName }),
+    ...(contact && { contact }),
+    ...(username && { username }),
+  };
+
+  console.log(userbody);
+  const [user, err] = await to(Admin.findOneAndUpdate({ email: res.locals.decode.email }, userbody));
```

Comment:
How do you plan to handle the case when the new username is already in use by someone else?
Suppose you want to change you Github username to jackfrost13. Is that possible? How will your code handle similar situation

Reply:
I don't know for now. But I'm looking into solving this issue.

Reply:
Sure, take you time.
Once you are done with everything re-request for a review.
You also need to share the API Testing files

Reply:
For checking if the username already exists, can I make another DB call?

Reply:
I suggest not to use another db call and to complete the entrire process using just one DB call